src/data_structures.rs

```diff
@@ -202,46 +201,41 @@ impl<C: PCCommitment> LabeledCommitment<C> {
     }
 
     /// Retrieve the commitment from `self`.
-    pub fn commitment(&self) -> &C {
+    pub fn commitment(&self) -> &G {
         &self.commitment
     }
 
-    /// Retrieve the degree bound in `self`.
-    pub fn degree_bound(&self) -> Option<usize> {
-        self.degree_bound
-    }
-
     /// Randomize commitment for test purpose
-    pub fn randomize(&mut self) {
-        self.commitment.randomize();
+    // #[cfg(test)]
+    pub fn ut_randomize(&mut self) {
+        self.commitment *= &G::ScalarField::rand(&mut rand::thread_rng())
     }
 }
 
-impl<C: PCCommitment> SemanticallyValid for LabeledCommitment<C> {
+impl<G: Group> SemanticallyValid for LabeledCommitment<G> {
     fn is_valid(&self) -> bool {
         self.commitment.is_valid()
     }
 }
 
-impl<C: PCCommitment> ToBytes for LabeledCommitment<C> {
+impl<G: Group> ToBytes for LabeledCommitment<G> {
```

Comment:
Totally agree, but this will take some time, because ToBytes trait is widely used not only in ginger-lib, marlin and poly-commit, but also in sc / mc / cctp libs. I haven't touched it because refactoring was focused on ginger and its API. Should I update all libs and replace this ToBytes trait everywhere by Cannonical ser/deser functions?

Reply:
I'm not speaking about deprecating it everywhere, but since you are touching this code let's do it at least for this one